fs/localfs/copy.go

```diff
@@ -0,0 +1,164 @@
+package localfs
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/fs"
+)
+
+// Copy copies e into targetPath in the local file system. If e is an
+// fs.Directory, then the contents are recursively copied.
+// The targetPath must not exist, except when the target path is the root
+// directory. In that case, e must be a fs.Directory and its contents are copied
+// to the root directory.
+// Copy does not overwrite files or directories and returns an error in that
+// case. It also returns an error when the the contents cannot be restored,
+// for example due to an I/O error.
+func Copy(ctx context.Context, targetPath string, e fs.Entry) error {
```

Comment:
possible future improvement: introduce abstraction for writing to an abstract filesystem, so we can restore to a zip/tar file too in addition to a local directory.